src/Schedule/RunContext.php

```diff
@@ -70,7 +65,7 @@ final public function getFormattedMemory(): string
 
     final protected function markAsRun(int $memory): void
     {
-        $this->duration = \time() - $this->startTime;
+        $this->duration = \time() - $this->startTime->getTimestamp();
```

Comment:
@kbond Wouldn't it be better to use `new \DateTimeImmutable('now')` instead of `\time()` here too, like in the constructor?

Reply:
My understanding is, when you call `->getTimestamp()`, it returns the "utc timestamp" which is what `time()` provides. Is that not the case?

This timezone stuff makes my head hurt...

Reply:
I'll try to explain better what I mean.

You changed `$this->startTime = \time();` to `$this->startTime = new \DateTimeImmutable('now');` in the constructor, because of the issue you mentioned in the PR description

Before you were comparing with `$this->duration = \time() - $this->startTime;)` here.

Now it is `$this->duration = \time() - $this->startTime->getTimestamp();`.

What I would suggest is `$this->duration = (new \DateTimeImmutable('now'))->getTimestamp() - $this->startTime->getTimestamp();` for consistency, so everything is based on `new \DateTimeImmutable('now')` instead of `time()`

Reply:
I can change for consistency but I believe the result would be the same(?)

I think the crux of the issue was getStartTime(). This took the start time timestamp and always converted to utc.